### PR TITLE
TextBlob common

### PR DESCRIPTION
**Pull request by eymar**
Even though CI still complains about some tests, I think this PR is ready for review. I'm fixing tests issues now.

Changes:
- Tried to reuse as much logic as possible: moved some classes and functions from jni to commonMain/cpp `mppinterop.cc`

### CODE REVIEW

File: .github/workflows/blank.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
           ./gradlew -Pskiko.native.enabled=true macosX64Test iosX64Test
           ./gradlew -Pskiko.native.enabled=true publishToMavenLocal
       - uses: actions/upload-artifact@v2
+        if: ${{ true }}
```
Comment (olonho) [inline .github/workflows/blank.yml:36]: Maybe remove those.

File: skiko/src/commonMain/cpp/common/include/RunRecordClone.hh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    uint8_t positioning() const {
+        return fFlags & 0x3; // kPositioning_Mask
+    }
+
+    bool isExtended() const {
+        return fFlags & 0x8; // kExtended_Flag
+    }
+
+    static unsigned ScalarsPerGlyph(uint8_t pos) {
+        const uint8_t gScalarsPerPositioning[] = {
```
Comment (olonho) [inline skiko/src/commonMain/cpp/common/include/RunRecordClone.hh:55]: Why we cannot do that with `switch`?

File: skiko/src/commonTest/kotlin/org/jetbrains/skia/TextBlobTest.kt [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+import kotlin.math.sin
+import kotlin.test.*
+
+class TextBlobTest {
+
+    private val inter36: suspend () -> Font = suspend {
+        Font(Typeface.makeFromResource("./fonts/Inter-Hinted-Regular.ttf"), 36f)
+    }
+
+    private val eps = 0.001f // only this value works for kotlin/js
```
Comment (olonho) [inline skiko/src/commonTest/kotlin/org/jetbrains/skia/TextBlobTest.kt:18]: Shall we select epsilon based on target platform?

File: .github/workflows/blank.yml (hunk 1 above)
Comment (eymar) [reply .github/workflows/blank.yml]: that was an attempt to make `upload-test-results` step run despite failing tests. But `${{true}}` didn't help for some reason. By default a step gets executed when preceding steps were successful. This issue doesn't let us download the test report when any test failed - not very useful. I'll check this problem later.

Removed for now.

File: skiko/src/commonMain/cpp/common/include/RunRecordClone.hh (hunk 2 above)
Comment (eymar) [reply skiko/src/commonMain/cpp/common/include/RunRecordClone.hh]: sure. Didn't think about it when copy/pasted from jni.

File: skiko/src/commonTest/kotlin/org/jetbrains/skia/TextBlobTest.kt (hunk 3 above)
Comment (eymar) [reply skiko/src/commonTest/kotlin/org/jetbrains/skia/TextBlobTest.kt]: applied your suggestion

**Pull request merged**